File: JS(Unused)/stack.js
export class Stack {
  #items = [];

  push(item) {
    this.#items.push(item);
  }

  pop() {
    return this.#items.pop();
  }

  peek() {
    return this.#items[this.#items.length - 1];
  }

  get isEmpty() {
    return this.#items.length === 0;
  }

  get size() {
    return this.#items.length;
  }

  toArray() {
    return [...this.#items];
  }
}

File: Python(DSA)/stack.py
# class ActionHistoryStack:
# 	def __init__(self):
# 		self.stack = []
		
# 	def push_action(self, action_string):
# 		self.stack.append(action_string)
# 		pushAction(action_string)

# 	def pop_action(self):
# 		if self.is_empty():
# 			raise IndexError("Popped from Empty Stack")
# 		return self.stack.pop()
	
# 	def peek_action(self):
# 		if self.is_empty():
# 			raise IndexError("Peeked from Empty Stack")
# 		return self.stack[-1]
		
# 	def is_empty (self):
# 		return len(self.stack) == 0
		
# stack = ActionHistoryStack()
# stack.push_action("You planted a Plant")
# stack.push_action("You watered your plant today")
	
# print("Action History Stack is ready!")
